lib/models/shard.js

```diff
@@ -36,90 +41,41 @@ ShardSchema.set('toObject', {
   transform: function(doc, ret) {
     delete ret.__v;
     delete ret._id;
-
-    doc.contracts = doc.contracts || [];
-    doc.trees = doc.trees || [];
-    doc.challenges = doc.challenges || [];
-    doc.meta =  doc.meta || [];
-
-    ret.contracts = {};
-    ret.trees = {};
-    ret.challenges = {};
-    ret.meta = {};
-
-    for (let i = 0; i < doc.contracts.length; i++) {
-      ret.contracts[doc.contracts[i].nodeID] = doc.contracts[i].contract;
-    }
-
-    for (let i = 0; i < doc.trees.length; i++) {
-      ret.trees[doc.trees[i].nodeID] = doc.trees[i].tree;
-    }
-
-    for (let i = 0; i < doc.challenges.length; i++) {
-      ret.challenges[doc.challenges[i].nodeID] = doc.challenges[i].challenge;
-    }
-
-    for (let i = 0; i < doc.meta.length; i++) {
-      ret.meta[doc.meta[i].nodeID] = doc.meta[i].meta;
-    }
```

Comment:
Yeah, the `storj.Contract` objects have these fields keyed by the nodeId of the other party, but to my knowledge defining a schema where the keys are variable is not possible/supported/desirable. Storing these parts in mongo this way (as an array) lets us query on them - but to be turned back into a contract object they have to use the nodeid-value format.